src/uct/cuda/base/cuda_iface.h

```diff
@@ -99,15 +99,19 @@
     UCT_CUDADRV_FUNC(_func, UCS_LOG_LEVEL_ERROR)
 
 
-#define UCT_CUDADRV_CTX_ACTIVE(_state) \
-    { \
-        CUcontext _ctx; \
-        if (CUDA_SUCCESS == cuCtxGetCurrent(&_ctx)) { \
-            _state = (_ctx == NULL) ? 0 : 1; \
-        } else { \
-            _state = 0; \
-        } \
-    }
+static UCS_F_ALWAYS_INLINE int uct_cuda_base_is_context_active()
+{
+    CUcontext ctx;
+
+    return (CUDA_SUCCESS == cuCtxGetCurrent(&ctx)) && (ctx != NULL);
+}
+
+
+static UCS_F_ALWAYS_INLINE int uct_cuda_base_context_match(CUcontext *ctx1,
+                                                           CUcontext *ctx2)
+{
+    return ((*ctx1 == NULL) && (*ctx1 == *ctx2));
```

Comment:
1. pass CUcontext by value not by pointer
2.  why we require ctx1==NULL? maybe it should be ctx1!=NULL?

Reply:
1. Changed to pass by value
2. is a mistake on my end. I've fixed this now.